tests/lib/prepare-restore.sh

```diff
@@ -543,6 +553,13 @@ restore_suite_each() {
         rm -rf "${PWD}.tar"
     fi
 
+    # The core18 snap may have been automatically installed by the test.
+    # If it was not present before the test then it should be removed.
+    if [ -e /run/spread/please-remove-core18 ]; then
```

Comment:
should this maybe be:

```suggestion
    if [ -f /run/spread/please-remove-core18 ]; then
```

I don't know why this file would "exist" but not be a regular file, but -f seems more natural to me.